pkg/ovncontroller/network.go

```diff
@@ -15,82 +15,184 @@ package ovncontroller
 import (
 	"context"
 	"fmt"
+	"slices"
+	"unicode"
 
 	netattdefv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
 	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	"golang.org/x/exp/maps"
 	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	ovnv1 "github.com/openstack-k8s-operators/ovn-operator/api/v1beta1"
 )
 
-// CreateOrUpdateAdditionalNetworks - create or update network attachment definitions based on the provided mappings
-func CreateOrUpdateAdditionalNetworks(
```

Comment:
Let's also have kuttl tests to validate this new config, can extend test ovn_nicmappings itself